CMakeLists.txt

```diff
@@ -475,6 +475,46 @@ endif()
 
 llvm_canonicalize_cmake_booleans(CIRCT_LLHD_SIM_ENABLED)
 
+#-------------------------------------------------------------------------------
+# circt-lec Configuration
+#-------------------------------------------------------------------------------
+# If circt-lec hasn't been explicitly disabled, find it.
+option(CIRCT_LEC_DISABLE "Disable the Logical Equivalence Checker" OFF)
+if(CIRCT_LEC_DISABLE)
+  message(STATUS "Disabling circt-lec")
+else()
+  if(Z3_DIR)
+    # Search and load the package configuration file in the specified directory.
+    find_package(Z3 CONFIG REQUIRED PATHS ${Z3_DIR} NO_DEFAULT_PATH)
+    if(Z3_FOUND)
+      # Report the found library location and version
+      # similarly to LLVM's `FindZ3` CMake module.
+      get_target_property(Z3_LIB_LOCATION z3::libz3 IMPORTED_LOCATION_DEBUG)
+      message(STATUS "Found Z3: ${Z3_LIB_LOCATION} (found version \"${Z3_VERSION_STRING}\")")
+    endif()
+    # Link against the reported library and include locations.
+    list(APPEND CIRCT_LEC_LIBS ${Z3_LIBRARIES})
+    list(APPEND CIRCT_LEC_INCLUDES ${Z3_CXX_INCLUDE_DIRS})
+  else()
+    # Attempt initialising Z3 according to LLVM's `FindZ3` CMake module.
+    find_package(Z3)
+    # Link against the reported library and include locations.
+    list(APPEND CIRCT_LEC_LIBS ${Z3_LIBRARIES})
+    list(APPEND CIRCT_LEC_INCLUDES ${Z3_INCLUDE_DIR})
+  endif()
+  
+  if(Z3_FOUND)
+    SET(CIRCT_LEC_Z3_VER 4.8.11)
+    if(Z3_VERSION VERSION_LESS ${CIRCT_LEC_Z3_VER})
+      message(WARNING "Cannot build circt-lec with outdated Z3 version ${Z3_VERSION}, requires ${CIRCT_LEC_Z3_VER}.")
```

Comment:
@frog-in-the-well Looks to me like there's an issue here - on my machine the build fails with z3 4.11.2. The problem seems to be that the variable `Z3_VERSION` is empty - both on my machine and in CI `circt-lec` doesn't build, and the message `Cannot build circt-lec with outdated Z3 version , requires 4.8.11.` is shown, with no outdated version number given. Does this need to be `Z3_VERSION_STRING`?

Reply:
@TaoBi22 Thank you for the catch, it should be working now.
I didn't expect that to not be unset from https://github.com/llvm/llvm-project/blob/d6ab9cae4b689b51ad2c62ff5adce8e8f11c527b/llvm/cmake/modules/FindZ3.cmake#L124